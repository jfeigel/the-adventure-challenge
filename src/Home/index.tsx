import React from 'react';

import { Box } from '@mui/material';

import logo from 'images/logo.png';

function Home() {
  return (
    <Box display="flex" justifyContent="center" alignItems="center" flex="1">
      <img src={logo} alt="Adventure Challenge logo" />
    </Box>
  );
}

export default Home;
