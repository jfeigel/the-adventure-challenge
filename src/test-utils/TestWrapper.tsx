import { ReactNode } from 'react';

import { render as jestRender } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import ThemeProvider from 'mui/components/ThemeProvider';

type Props = {
  children?: ReactNode;
};

function TestWrapper(props: Props) {
  return <ThemeProvider>{props.children}</ThemeProvider>;
}

export function render(component: ReactNode) {
  const wrapped = <TestWrapper>{component}</TestWrapper>;
  return jestRender(wrapped, { wrapper: MemoryRouter });
}
