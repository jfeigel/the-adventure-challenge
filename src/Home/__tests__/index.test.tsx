import { render } from 'test-utils/TestWrapper';

import Home from 'Home';

describe('Home tests', () => {
  it('Should render the home page', () => {
    const { container } = render(<Home />);

    expect(container).toMatchSnapshot();
  });
});
