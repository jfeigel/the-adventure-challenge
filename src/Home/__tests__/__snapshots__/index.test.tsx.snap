// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home tests Should render the home page 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-znd6jn-MuiContainer-root"
  >
    <img
      alt="Adventure Challenge logo"
      class="MuiBox-root css-1vunpsv"
      src="logo.png"
    />
  </div>
</div>
`;
